### ISSUE TITLE
GamePlayers and GamePlayer should not be nodes

### ISSUE THREAD

**Issue by SamTheBlow** [SamTheBlow/grand-strategy-game#229]
They're part of the game logic, and so they shouldn't have anything to do with nodes.

The only reason why they're nodes is because GamePlayer synchronizes its username changes.
So let's synchronize username changes in a separate class.
First, let's remember that human player usernames are already sync'd in Player, and GamePlayer's username is the Player username for human players, so all we want is to sync usernames for AI players.
We need a Node that is in the scene tree for as a long as the game exists. Let's name it GameUsernameSync. It needs to listen to username change in all game players. When it happens, if the user is the server, it sends the change to clients. Then clients receive the username change and apply the change.

- [x] There's a new node GameUsernameSync
- [x] It enters the scene tree whenever a game is created
- [x] It leaves the scene tree whenever a game is discarded
- [x] It takes a GamePlayers object
- [x] GamePlayers and GamePlayer now have a "username_changed" signal that passes a GamePlayer
- [x] When it's given a GamePlayers object, it listens to its "username_changed" signal
- [x] When the signal is emitted, if the user is the server, it sends the change to all clients
- [x] All clients receive the username change and apply the change locally

**Comment by SamTheBlow** [comment]
This was done in commit 532dc69, however, there are some problems:
- Whenever a username changes after GameUsernameSync is added on the server, but before it's added on the client, an error occurs and the client never receives the change. This can be circumvented by avoiding username changes during game load, so whatever.
- When a client joins a server whose game already started, the server adds the client's players and assigns them to existing gameplayers, which in turn (I think) sets the username property of the new gameplayers which in turn triggers the server's GameUsernameSync before the client created the node, and bam, the error I mentioned above. In the end, it doesn't matter, because the client receives all the correct usernames when receiving the game, so no information is lost.

TL;DR there is no problem lol
